src/Symfony/Component/Process/Process.php

```diff
@@ -468,7 +468,7 @@ public function getIncrementalErrorOutput()
      */
     public function getExitCode()
     {
-        if ($this->isSigchildEnabled() && !$this->enhanceSigchildCompatibility) {
+        if (!$this->enhanceSigchildCompatibility && $this->isSigchildEnabled()) {
```

Comment:
Didn't take into account emerging issues, but you absolutely right here. Will revert this and according changes in Processes.